src/scripts/cronjobs/firstDataBrokerRemovalFixed.tsx

```diff
@@ -0,0 +1,124 @@
+import { sanitizeSubscriberRow } from "../../app/functions/server/sanitize";
+import { refreshStoredScanResults } from "../../app/functions/server/refreshStoredScanResults";
+import { getLatestOnerepScanResults } from "../../db/tables/onerep_scans";
+
+type SubscriberFirstRemovedScanResult = {
+  subscriber: SubscriberRow;
+  firstRemovedScanResult: OnerepScanResultRow;
+};
+
+function isFulfilledResult(
+  result: PromiseSettledResult<SubscriberFirstRemovedScanResult | undefined>,
+): result is PromiseFulfilledResult<SubscriberFirstRemovedScanResult> {
+  return typeof result !== "undefined" && result.status === "fulfilled";
+}
+
+void run();
+
+async function run() {
+  const batchSize = Number.parseInt(
+    process.env.FIRST_DATA_BROKER_REMOVAL_FIXED_EMAIL_BATCH_SIZE ?? "10",
+    10,
+  );
+  if (Number.isNaN(batchSize)) {
+    throw new Error(
+      `Could not send first data broker removal fixed emails, because the env var FIRST_DATA_BROKER_REMOVAL_FIXED_EMAIL_BATCH_SIZE has a non-numeric value: [${process.env.FIRST_DATA_BROKER_REMOVAL_FIXED_EMAIL_BATCH_SIZE}].`,
+    );
+  }
+  const potentialSubscribersToEmail =
+    await getPotentialSubscribersWaitingForFirstDataBrokerRemovalFixedEmail();
+
+  const subscribersToEmailWithData = (
+    await Promise.allSettled(
+      potentialSubscribersToEmail.map(async (subscriber) => {
+        // OneRep suggested not relying on webhooks, but instead to fetch the latest
+        // data from their API. Thus, let's refresh the data in our DB in real-time:
+        if (subscriber.onerep_profile_id !== null) {
+          await refreshStoredScanResults(subscriber.onerep_profile_id);
+        }
+        const latestScan = await getLatestOnerepScanResults(
+          subscriber.onerep_profile_id,
+        );
+
+        let firstRemovedScanResult = null;
+        for (const scanResult of latestScan.results) {
+          if (
+            scanResult.manually_resolved &&
+            scanResult.status === "removed" &&
+            (!firstRemovedScanResult ||
+              (firstRemovedScanResult &&
+                scanResult.created_at.getTime() <
+                  firstRemovedScanResult.created_at.getTime()))
+          ) {
+            firstRemovedScanResult = scanResult;
+          }
+        }
+
+        if (!firstRemovedScanResult) {
+          return;
+        }
+
+        return { subscriber, firstRemovedScanResult };
+      }),
+    )
+  )
+    .filter(isFulfilledResult)
+    .slice(0, batchSize);
+
+  await initEmail();
+
+  await Promise.allSettled(
+    subscribersToEmailWithData.map((data) => {
+      return sendFirstDataBrokerRemovalFixedActivityEmail(
+        data.value.subscriber,
+        data.value.firstRemovedScanResult,
+      );
+    }),
+  );
+  console.log(
+    `[${new Date(Date.now()).toISOString()}] Sent [${subscribersToEmailWithData.length}] first data broker removal fixed emails.`,
+  );
+}
+
+async function sendFirstDataBrokerRemovalFixedActivityEmail(
```

Comment:
For example if this function was the only unique thing for each email cron job I think that would be very easy to understand!